msteams-platform/m365-apps/teamsjs-support-m365.md

```diff
@@ -0,0 +1,2069 @@
+
+Starting with version 2.0.0, Teams JavaScript client library (TeamsJS) enables [certain types of Teams apps](./overview.md) to run across the Microsoft 365 ecosystem. Currently, Microsoft 365 application hosts (Microsoft 365 app and Outlook) for Teams apps support a subset of the application types and capabilities you can build for the Teams platform. This support will expand over time.
+
+This article details the level of support of Teams JavaScript client (TeamsJS) library version 2.x capabilities for *personal scope* tab apps and dialogs (task modules) across various Microsoft 365 host applications. For a more info on what's changed between TeamsJS versions 1.x and 2.x, see [What's new in TeamsJS version 2.x.x](../tabs/how-to/using-teams-client-library.md#whats-new-in-teamsjs-version-2xx).
+
+> [!TIP]
+> You can check for host support of a given capability at runtime by calling the `isSupported()` function on that capability (namespace or subnamespace).
+
+The *TeamsJS Capability* table below lists TeamsJS capabilities (public namespaces) and their support across Microsoft 365 hosts. Some capabilties are marked with  "Deprecated" or "Preview" badges, which have the following meanings:
+
+| Badge | Meaning |
+| - | - |
+| Preview :::image type="content" source="./images/preview-badge.png" alt-text="Image with the word 'Preview' inside a blue rectangle"::: | This capability is in preview and subject to change based on feedback. Please do not use in production. |
+| Deprecated :::image type="content" source="./images/deprecated-badge.png" alt-text="Image with the word 'Deprecated' inside an orange rectangle"::: | This capability is deprecated in favor of newer functionality, though  supported for backwards compability purposes. For new apps, please use the capability recommended in the usage notes of the deprecated capability. |
+
+For readability, Microsoft 365 hosts are signified by the following product icons:
+
+| Microsoft Teams | Microsoft 365 app | Microsoft Outlook |
+| :-: | :-: | :-: |
+| :::image type="content" source="./images/teams-icon.png" alt-text="Microsoft Teams icon"::: | :::image type="content" source="./images/microsoft-365-icon.png" alt-text="Microsoft 365 app icon"::: | :::image type="content" source="./images/outlook-icon.png" alt-text="Microsoft Outlook icon"::: |
+
+Entries marked with a check and asterisk (&#x2713;*) indicate support for that host is available only to preview audiences. Click on any TeamsJS Capability to find further details, including host version availability, recommended alternative capability or APIs (for deprecated capabilities), usage notes, and further resources.
+
+<br />
+<table border>
+    <thead>
+        <tr>
+            <th>TeamsJS Capability</th>
+            <th colspan=3>Web</th>
+            <th colspan=3>Desktop</th>
+            <th colspan=6>Mobile</th>
+        </tr>
+        <tr>
+            <th></th>
+            <th colspan=3></th>
+            <th colspan=3>Windows</th>
```

Comment:
>Windows [](http://example.com/codeflow?start=26&length=7)

Not to add complexity to this, but if we want to be complete, do we also have to list Teams for Mac?

Reply:
Hmm, fair question. I'd argue that'd add a whole lot of blank cells when the matrices as-is have a lot of coverage to fill in. Also of limited value to customers until there's any earnest multi-hub story for Mac experiences. @vikramtha what do you think?